pages/index.js

```diff
@@ -1,13 +1,26 @@
 import Head from "next/head";
 import RoastCard from "@/components/RoastCard";
 import { StyledList, StyledItem } from "@/lib/styled-components";
+import { useSession } from "next-auth/react";
 import useSWR from "swr";
 
 const fetcher = (...args) => fetch(...args).then((res) => res.json());
 
 export default function Home() {
+  const { data: session } = useSession();
   const { data, isLoading } = useSWR("api/roasts", fetcher);
-  if (isLoading) return <h1>... is Loading</h1>;
+  const { data: favorites, isLoading: favoritesLoading } = useSWR(
+    () => (session ? "api/user" : null),
+    fetcher
+  );
+  if (isLoading || favoritesLoading) return <h1>... is Loading</h1>;
+
+  function isFavorite(id) {
+    if (session) {
+      return favorites.roasts.some((roast) => roast._id === id);
+    } else return false;
+  }
```

Comment:
You could simplify this function as well if you like.

Reply:
How? It needs the function for each RoastCard to check if the ID is also in Favorites.